internal/support/scheduler/v2/application/scheduler/queue.go

```diff
@@ -0,0 +1,209 @@
+//
+// Copyright (C) 2021 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package scheduler
+
+import (
+	"fmt"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/errors"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/v2/models"
+)
+
+func (m *manager) addInterval(interval models.Interval, sc *Executor) {
+	m.intervalToExecutorMap[interval.Name] = sc
+	m.executorQueue.Add(sc)
+}
+
+func (m *manager) deleteInterval(sc *Executor) {
+	delete(m.intervalToExecutorMap, sc.Interval.Name)
+	// Mark as Deleted and scheduler will remove it from the queue
+	sc.MarkedDeleted = true
+}
+
+func (m *manager) addIntervalAction(sc *Executor, action models.IntervalAction) {
+	sc.IntervalActionsMap[action.Name] = action
+	m.actionToIntervalMap[action.Name] = sc.Interval.Name
+}
+
+func (m *manager) IntervalByName(intervalName string) (models.Interval, errors.EdgeX) {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+
+	sc, exists := m.intervalToExecutorMap[intervalName]
+	if !exists {
+		return models.Interval{},
+			errors.NewCommonEdgeX(errors.KindContractInvalid,
+				fmt.Sprintf("the executor with name %s does not exist", intervalName), nil)
+	}
+	return sc.Interval, nil
+}
+
+func (m *manager) AddInterval(interval models.Interval) errors.EdgeX {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+
+	if _, exists := m.intervalToExecutorMap[interval.Name]; exists {
+		return errors.NewCommonEdgeX(errors.KindContractInvalid,
+			fmt.Sprintf("the executor with name : %s already exists", interval.Name), nil)
+	}
+
+	executor := Executor{
+		IntervalActionsMap: make(map[string]models.IntervalAction),
+		MarkedDeleted:      false,
+	}
+	err := executor.Initialize(interval, m.lc)
+	if err != nil {
+		return errors.NewCommonEdgeXWrapper(err)
+	}
+	m.addInterval(interval, &executor)
+
+	m.lc.Infof("added the interval with name : %s into the scheduler queue", interval.Name)
+	return nil
+}
+
+func (m *manager) UpdateInterval(interval models.Interval) errors.EdgeX {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+
+	executor, exists := m.intervalToExecutorMap[interval.Name]
+	if !exists {
+		return errors.NewCommonEdgeX(errors.KindContractInvalid,
+			fmt.Sprintf("the executor with name %s does not exist", interval.Name), nil)
+	}
+	err := executor.Initialize(interval, m.lc)
+	if err != nil {
+		return errors.NewCommonEdgeXWrapper(err)
+	}
+	m.lc.Infof("updated the interval with name: %s in the scheduler queue", interval.Name)
+	return nil
+}
+
+func (m *manager) DeleteIntervalByName(intervalName string) errors.EdgeX {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+	m.lc.Debugf("removing the interval with name: %s ", intervalName)
+
+	sc, exists := m.intervalToExecutorMap[intervalName]
+	if !exists {
+		return errors.NewCommonEdgeX(errors.KindContractInvalid,
+			fmt.Sprintf("the executor with name %s does not exist", intervalName), nil)
+	}
+	m.deleteInterval(sc)
+
+	m.lc.Infof("removed the interval with id: %s from the scheduler queue", intervalName)
```

Comment:
Change `removed the interval with id` => `removed the interval with name`

Reply:
Modified.